### PR TITLE
Update EIP-3643: Move to Draft

### PR DESCRIPTION
**Pull request by Joachim-Lebrun**
- put EIP-3643 out of stagnant status
- previous status was draft, so the current PR sets the status back to draft, following recommandations of EIP-1
- Adapt the content of the EIP to the last audited version of the T-REX protocol
- Remove useless reference to upgradeable proxies => proxies are used in T-REX but it is up to the developer to decide if they want to make the token upgradeable or not. as long as interfaces are matching the standard it doesn't matter. 
- Remove duplicate functions, e.g. `transferOwnershipOnIdentityRegistryContract` which is a duplicate of `transferOwnership` as we mention the ERC-173 used by the standard. 
- Remove all links to external sources, assets added in the repository (pdf files of audits/interface contracts) 
- Remove references to ERC-734 and ERC-735 as these are not official EIPs
- Fix order of sections and other elements according to EIP-1 
- add reference to ERC-173 instead of implementation recommandation (OZ Ownable library) 
- add the interface of IAgentRole as the role is important in the standard, previously it was described with implementation related reference, now we focus on the externally visible functions of contracts only, not implementation details

### CODE REVIEW

File: assets/eip-3643/TREX_V4_audit.pdf
Comment (SamWilsn) [inline assets/eip-3643/TREX_V4_audit.pdf:1]: Please remove your audit results.

File: EIPS/eip-3643.md [hunk 1]
```diff
@@ -36,7 +32,7 @@ The functions described by these interfaces vary and allow the appropriate users
 The following requirements have been compiled following discussions with parties across financial institutions that are looking to issue securities on a DLT infrastructure such as ethereum.
 
 - **MUST** be [ERC-20](./eip-20.md) compatible.
-- **MUST** be used in combination with an Identification system onchain ([ONCHAINID](https://github.com/onchain-id/solidity))
+- **MUST** be used in combination with an onchain Identity system
 - **MUST** be able to apply any rule of compliance that is required by the regulator or by the token issuer (about the factors of eligibility of an identity or about the rules of the token itself)
 - **MUST** have a standard interface to pre-check if a transfer is going to pass or fail before sending it to the blockchain
 - **MUST** have a recovery system in case an investor loses access to his private key
```
Comment (SamWilsn) [inline EIPS/eip-3643.md:38]: Because this isn't the specification section, you can't add requirements on implementations here. From my quick reading, it seems like these are requirements on the standard itself? To avoid confusion, I'd recommend not using the capitalized `MUST` notation, and maybe rewording this into a paragraph instead of bullet points.

File: EIPS/eip-3643.md [hunk 2]
```diff
@@ -181,21 +163,28 @@ interface IERC3643 is IERC20 {
...
-
-Note that [`IClaimIssuer`](https://github.com/onchain-id/solidity/blob/master/contracts/interface/IClaimIssuer.sol) and [`IIdentity`](https://github.com/onchain-id/solidity/blob/master/contracts/interface/IIdentity.sol) are needed in this interface and are coming from [ONCHAINID](https://github.com/onchain-id/solidity)
+This Identity Registry is linked to storage that contains a dynamic whitelist of identities. The Identity Registry 
+makes the link between a wallet address, an Identity smart contract and a country code corresponding to the country of 
+residence of the investor, this country code is set in accordance with the ISO-3166 standard. It also contains a 
+function called `isVerified()`, which returns a status based on the validity of claims (as per the security token 
+requirements) in the user’s Identity contract.
+In the T-REX implementation of this standard, we decided to use the Ownable library from OpenZeppelin as well as an 
+Agent Role implemented with the [Roles Library](../assets/eip-3643/roles/Roles.sol) to manage access to functions on
```
Comment (SamWilsn) [inline EIPS/eip-3643.md:178]: EIPs should only specify the externally visible properties of a smart contract. They should not specify _how_ an implementation implements those properties.

File: assets/eip-3643/TREX_V3_audit.pdf
[the remark this answers is not in the document]
Comment (Joachim-Lebrun) [reply assets/eip-3643/TREX_V3_audit.pdf:1]: done in https://github.com/ethereum/EIPs/pull/7164/commits/3bf9e92260f046a89ca38fc19361d6ee610d3a67

File: assets/eip-3643/TREX_V4_audit.pdf
Comment (Joachim-Lebrun) [reply assets/eip-3643/TREX_V4_audit.pdf:1]: done in https://github.com/ethereum/EIPs/pull/7164/commits/3bf9e92260f046a89ca38fc19361d6ee610d3a67

File: EIPS/eip-3643.md (hunk 2 above)
Comment (Joachim-Lebrun) [reply EIPS/eip-3643.md]: done in https://github.com/ethereum/EIPs/pull/7164/commits/97819ae191f6b2a9c9feaeb92c8d35214853b2de 
I removed all mentions of implementation, preferring the reference to ERC-173 for ownership, i added the interface `IAgentRole` to the specifications as the agent role is very crucial for the standard. A more granular approach of the Agent Role can be done through an external contract for better role management but it seems important to define the 2 main roles, `owner` and `agent`

File: EIPS/eip-3643.md [hunk 3]
```diff
@@ -181,21 +163,28 @@ interface IERC3643 is IERC20 {
...
-A detailed description of the functions can be found in the [interfaces folder](https://github.com/TokenySolutions/EIP3643/tree/main/interfaces). 
-
-Note that [`IClaimIssuer`](https://github.com/onchain-id/solidity/blob/master/contracts/interface/IClaimIssuer.sol) and [`IIdentity`](https://github.com/onchain-id/solidity/blob/master/contracts/interface/IIdentity.sol) are needed in this interface and are coming from [ONCHAINID](https://github.com/onchain-id/solidity)
+This Identity Registry is linked to storage that contains a dynamic whitelist of identities. The Identity Registry 
+makes the link between a wallet address, an Identity smart contract and a country code corresponding to the country of 
+residence of the investor, this country code is set in accordance with the ISO-3166 standard. It also contains a 
+function called `isVerified()`, which returns a status based on the validity of claims (as per the security token 
+requirements) in the user’s Identity contract.
+In the T-REX implementation of this standard, we decided to use the Ownable library from OpenZeppelin as well as an
```
[the remark this answers is not in the document]
Comment (Joachim-Lebrun) [reply EIPS/eip-3643.md]: all implementation recommandations have been removed in https://github.com/ethereum/EIPs/pull/7164/commits/97819ae191f6b2a9c9feaeb92c8d35214853b2de 
Instead i focused on the externally visible properties of contracts

File: EIPS/eip-3643.md [hunk 4]
```diff
@@ -363,19 +368,46 @@ interface IClaimTopicsRegistry {
...
+
+The T-REX standard supports the implementation of additional compliance rules through modular compliance. These modules can be used to enforce a wide range of rules and restrictions, such as caps on the number of investors or the percentage of tokens held by a single investor, restrictions on transfers between certain types of investors, and more. This flexibility allows issuers to tailor the compliance rules of their tokens to their specific needs and regulatory environment.
+
+## Backwards Compatibility
+
+T-REX tokens should be backwards compatible with ERC-20 and ERC-173
+and should be able to interact with a [Claim Holder contract](../assets/eip-3643/ONCHAINID/IERC735.sol) to validate 
+the claims linked to an [Identity contract](../assets/eip-3643/ONCHAINID/IIdentity.sol).
+
 ## Test Cases
```
Comment (SamWilsn) [inline EIPS/eip-3643.md:404]:
```suggestion
```

File: EIPS/eip-3643.md [hunk 5]
```diff
@@ -363,19 +368,46 @@ interface IClaimTopicsRegistry {
...
 
-The standard is implemented and tested with full coverage on Tokeny's [T-REX repository](https://github.com/TokenySolutions/T-REX)
+The standard is implemented and tested with full coverage on Tokeny's T-REX repository.
```
Comment (SamWilsn) [inline EIPS/eip-3643.md:405]:
```suggestion
```

File: EIPS/eip-3643.md [hunk 6]
```diff
@@ -363,19 +368,46 @@ interface IClaimTopicsRegistry {
...
 
 ## Security Considerations
 
-The suite of Smart Contracts has been audited by an external and independent company. The results can be found in [this document](https://tokeny.com/wp-content/uploads/2020/05/Tokeny-Solutions_T-REX-v3_Smart-Contract-Audit-Report_Kapersky.pdf).
+The suite of Smart Contracts has been audited by external independent companies. The results can be found on 
+Tokeny's website or on the Tokeny's T-REX repository.
```
Comment (SamWilsn) [inline EIPS/eip-3643.md:410]: Mentioning your website is conceptually an "external link", and so isn't allowed. I'd recommend either summarizing the results of the audit, or simply noting that no security issues have been found.

Perhaps something like one of these, as appropriate:

---

> This specification has been audited by Kapersky, and no notable security considerations were found.

---

> An audit, performed by Kapersky, highlighted the following considerations for implementers of this standard:
>  * When flooping a glarp, implementers must pay particular attention to the meep. It must remain glooped for the duration of the merp.
>  * Never transfer the moop to the merp. Doing so would allow an attacker to floop the glarp too early.

---

From an _extremely_ quick read of the audit report, it looks like it was focused on your implementation, but **not on this standard itself**. In other words, they focused on whether your implementation conformed to your whitepaper, and didn't consider whether your overall design was sound.

That isn't problematic in and of itself, but it does mean the audit might not be applicable to other implementers?

Comment (Joachim-Lebrun) [reply EIPS/eip-3643.md:407]: Indeed, the audit mainly focused on the concrete implementation of the standard. However, it is important to note that our implementation is a direct translation of the standard specification outlined in the whitepaper. Therefore, the audit process essentially included a thorough review and verification of the standard's design and functionality.

While the audit report may not explicitly state the details of these discussions, the audit process was comprehensive and spanned over four months. During this time, the audit team harshly challenged the principles and design decisions underlying the T-REX standard (ERC-3643). Their inquiry was not limited to the security aspects of smart contracts, but also delved into the rationale behind the standard's specifications. Satisfactory responses to these inquiries and the absence of any outstanding safety issues in the audit report demonstrate the robustness of the standard.

It is true that the audits we realized may not be fully applicable to other implementations of the standard, but still provide valuable insight into the standard's design and functionality. The audits validate the standard's specifications and provide assurance that the standard does not present any notable security issues (when implemented as per the specifications). Therefore, we consider it relevant and useful to mention the audits in this section related to standard security considerations.

Comment (Joachim-Lebrun) [reply EIPS/eip-3643.md]: Updated the `Security Considerations` section in https://github.com/ethereum/EIPs/pull/7164/commits/adfc130d6dd1ed41fd86709401c7ed93b8d6178c

**Review by SamWilsn — APPROVED** [review]

**Comment by Joachim-Lebrun** [comment]
@SamWilsn is there still any changes required to merge this PR?
